### PR TITLE
Fix alignment of user menu avatar

### PR DESCRIPTION
**Pull request by dbkr**
https://github.com/matrix-org/matrix-react-sdk/pull/12267 made the avatar a couple of pixels out of alignment with the space icons. It just needed to be moved 2px to the right to match the 18px margin of the space icons (including the padding from the black border).

Also change elements in the user menu to block elements so they don't generate extra vertical space in the user menu div, which was causing the hairline to be too far down (unless anyone knows of a better way to avoid this extra height in the div: setting the hight of the div explicitly didn't work).

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

### CODE REVIEW

File: res/css/structures/_SpacePanel.pcss [hunk 1]
```diff
@@ -390,8 +390,17 @@ limitations under the License.
...
-        margin: 12px 14px 4px 16px;
+        margin: 12px 14px 4px 18px;
+        width: min-content;
         max-width: 226px;
+
+        /* Display the container and img ehere as block elements so they don't take
+         * up extra vertical space.
+         */
+        span,
+        img {
```
Comment (robintown) [inline res/css/structures/_SpacePanel.pcss:401]: Could you target a class name here so we're not relying on the specific elements used in the Compound Avatar implementation?

File: res/css/structures/_SpacePanel.pcss [hunk 2]
```diff
@@ -390,8 +390,17 @@ limitations under the License.
     .mx_UserMenu {
         padding-bottom: 12px;
         border-bottom: 1px solid $separator;
-        margin: 12px 14px 4px 16px;
+        margin: 12px 14px 4px 18px;
+        width: min-content;
         max-width: 226px;
+
+        /* Display the container and img ehere as block elements so they don't take
```
Comment (robintown) [inline res/css/structures/_SpacePanel.pcss:397]:
```suggestion
        /* Display the container and img here as block elements so they don't take
```

File: res/css/structures/_SpacePanel.pcss (hunk 1 above)
Comment (dbkr) [reply res/css/structures/_SpacePanel.pcss]: I can for the span, but the image only has an autogenerated class name. Unsure whether adding one just for this or just using the element type is the lesser evil. We could also match everything, which might actually be better?

Comment (robintown) [reply res/css/structures/_SpacePanel.pcss]: From my brief testing it looked like setting `display: block` only on the span was enough to fix the height, for what it's worth

**Review by robintown — APPROVED** [review]

File: res/css/structures/_SpacePanel.pcss (hunk 1 above)
Comment (dbkr) [reply res/css/structures/_SpacePanel.pcss]: Huh, I retested and you're absolutely right.

**Pull request merged**